libcxx/include/__utility/small_buffer.h

```diff
@@ -66,16 +66,14 @@ class __small_buffer {
     if constexpr (__fits_in_buffer<_Stored>) {
       return std::launder(reinterpret_cast<_Stored*>(__buffer_));
     } else {
-      byte* __allocation = static_cast<byte*>(::operator new[](sizeof(_Stored), align_val_t{alignof(_Stored)}));
-      std::construct_at(reinterpret_cast<byte**>(__buffer_), __allocation);
-      return std::launder(reinterpret_cast<_Stored*>(__allocation));
+      return static_cast<_Stored*>(std::__libcpp_allocate(_BufferSize * sizeof(byte), alignof(_Stored)));
```

Comment:
This change looks wrong to me. We want to store the pointer inside the `__small_buffer` and then return the pointer to the storage. Right now we're just throwing away the pointer to the storage, since the caller isn't supposed to save it.

Reply:
Ah, yes, my suggestion in https://github.com/llvm/llvm-project/pull/114396#discussion_r1828007471 was incorrect (it was based on the already incorrect diff). I think this is what we want (which is a mix between my suggestion and the status quo):

```c++
byte* __allocation = static_cast<byte*>(std::__libcpp_allocate(sizeof(_Stored), alignof(_Stored)));
std::construct_at(reinterpret_cast<byte**>(__buffer_), __allocation);
return std::launder(reinterpret_cast<_Stored*>(__allocation));
```

Reply:
Thank you Louis and Nikolas !!!